### PR TITLE
Fix output shape dimension for OnnxConfigWithPast

### PR DESCRIPTION
**Pull request by fxmarty**
Following https://github.com/huggingface/optimum/pull/764/files#diff-c8b74c7c590ff56a6ecd02908c5f4181afc53fa9bf513e0ac4c7b7c65fecc200L57 , the GPU tests started failing: https://github.com/huggingface/optimum/actions/runs/4139994852/jobs/7158229313

This is because when using CUDAExecutionProvider + IO Binding, the input dynamic axis name `sequence_length` was previously used to infer the axis `sequence_length` of logits that was wrongly registered as having a dynamic length in the case of the with-past model. This PR fixes the issue.

The tests
```
pytest tests/onnxruntime/test_modeling.py -k "test_pipeline_on_gpu and ORTModelForCausalLMIntegrationTest" -s
pytest tests/onnxruntime/test_modeling.py -k "test_compare_generation_to_io_binding and ORTModelForSeq2SeqLMIntegrationTest" -s
```

now go well.

### CODE REVIEW

File: optimum/onnxruntime/modeling_ort.py [hunk 1]
```diff
@@ -759,6 +759,7 @@ def _prepare_io_binding(
                 output_shape = []
                 for axis_name in output_node.shape:
                     output_shape.append(self._output_shape_inference(axis_name, dimensions))
+
```
Comment (fxmarty) [inline optimum/onnxruntime/modeling_ort.py:762]:
```suggestion
```
